dali/operators/decoder/video/video_decoder_base.h

```diff
@@ -71,20 +58,65 @@ class DLL_PUBLIC VideoDecoderBase : public Operator<Backend> {
     return shape;
   }
 
+  TensorListShape<4> ReadOutputShape(int nframes, int batch_size) {
+    // to do returns whole file shape, not subsequence
+    TensorListShape<4> shape(batch_size);
+    for (int s = 0; s < batch_size; ++s) {
+      TensorShape<4> sample_shape;
+      sample_shape[0] = nframes;
+      sample_shape[1] = frames_decoders_[s]->Height();
+      sample_shape[2] = frames_decoders_[s]->Width();
+      sample_shape[3] = frames_decoders_[s]->Channels();
+      shape.set_tensor_shape(s, sample_shape);
+    }
+    return shape;
+  }
+
+
   /**
    * @brief Decode sample with index `idx` to `output` tensor.
    */
   void DecodeSample(SampleView<OutBackend> output, int64_t idx) {
-    auto &frames_decoder = *frames_decoders_[idx];
     int64_t num_frames = output.shape()[0];
+    DecodeFrames(output, idx, 0, num_frames);
+  }
+
+
+  /**
+   * to do
+   * @param output
+   * @param sample_idx
+   * @param start_idx
+   * @param num_frames
+   */
+  void DecodeFrames(SampleView<OutBackend> output, int64_t sample_idx, int64_t start_idx,
+                    int64_t num_frames) {
+    auto &frames_decoder = *frames_decoders_[sample_idx];
     int64_t frame_size = frames_decoder.FrameSize();
     uint8_t *output_data = output.template mutable_data<uint8_t>();
-    for (int f = 0; f < num_frames; ++f) {
+    for (int64_t f = 0; f < num_frames; ++f) {
       frames_decoder.ReadNextFrame(output_data + f * frame_size);
     }
   }
 
+
+  /**
+   * to do
+   * @param sample_idx
+   * @return
+   */
+  bool CanDecode(int64_t sample_idx) {
```

Comment:
This could be tricky. Do you want to check if you can decode next frame? If so we need to decode that particular frame and store it somewhere.